pkg/server/auth/passthrough.go

```diff
@@ -35,5 +41,32 @@ func (pg *BearerTokenPassthroughPrincipalGetter) Principal(r *http.Request) (*Us
 		return nil, nil
 	}
 
-	return &UserPrincipal{Token: extractToken(token)}, nil
+	token = extractToken(token)
+
+	config, err := kubecfg.GetConfig()
```

Comment:
Maybe we pass in authSrv.cfg.kubernetesClient instead of creating a new client wdyt?

Reply:
I don't think they're the same client. `authSrv.cfg.kubernetesClient` is a `sigs.k8s.io/controller-runtime/pkg/client.Client`, whereas the client in `passthrough.go‘ is a `k8s.io/client-go/kubernetes.ClientSet`

Reply:
We also pass the AuthServer as a param, so we don't have access to the private fields on the auth server object. We could make them public, or add a method to return the client.